workspace-manager-clienttests/src/main/java/scripts/testscripts/PrivateControlledGcsBucketLifecycle.java

```diff
@@ -0,0 +1,272 @@
+    //   logger.info("Workspace owner cannot read from private bucket");
+    // }
+
+    // TODO(PF-624): workspace readers can read from private buckets via "roles/viewer".
+    // // Workspace reader also cannot read private bucket contents
+    // try {
+    //   workspaceReaderStorageClient.get(blobId);
+    //   throw new IllegalStateException("Workspace reader was able to read private bucket");
+    // } catch (StorageException storageException) {
+    //   assertThat(storageException.getCode(), equalTo(HttpStatusCodes.STATUS_CODE_FORBIDDEN));
+    //   logger.info("Workspace reader cannot read from private bucket");
+    // }
+
+    // Private resource user can delete the blob they created earlier.
+    privateUserStorageClient.delete(blobId);
+    logger.info("Private resource user successfully deleted blob {}", blobId.getName());
+
+    // Private resource user cannot delete bucket, as they are not an editor
+    ControlledGcpResourceApi privateResourceUserWsmClient =
+        ClientTestUtils.getControlledGcpResourceClient(privateResourceUser, server);
+    try {
+      deleteBucketAttempt(privateResourceUserWsmClient, resourceId);
+      throw new IllegalStateException("non-editor private resource user was able to delete bucket");
+    } catch (ApiException apiException) {
+      assertThat(apiException.getCode(), equalTo(HttpStatusCodes.STATUS_CODE_FORBIDDEN));
+      logger.info("Private resource user cannot to delete bucket");
+    }
+
+    // Owner can delete the bucket through WSM
+    var ownerDeleteResult = deleteBucketAttempt(resourceApi, resourceId);
+    logger.info(
+        "For owner, delete bucket status is {}",
+        ownerDeleteResult.getJobReport().getStatus().toString());
+    assertThat(
+        ownerDeleteResult.getJobReport().getStatus(), equalTo(JobReport.StatusEnum.SUCCEEDED));
+
+    // verify the bucket was deleted from WSM metadata
+    try {
+      resourceApi.getBucket(getWorkspaceId(), resourceId);
+      throw new IllegalStateException("Incorrectly found a deleted bucket!");
+    } catch (ApiException ex) {
+      assertThat(ex.getCode(), equalTo(HttpStatusCodes.STATUS_CODE_NOT_FOUND));
+    }
+
+    // also verify it was deleted from GCP
+    Bucket maybeBucket = ownerStorageClient.get(bucketName);
+    assertThat(maybeBucket, Matchers.is(Matchers.nullValue()));
+  }
+
+  private CreatedControlledGcsBucket createPrivateBucketAttempt(
+      ControlledGcpResourceApi resourceApi) throws Exception {
+    String jobId = UUID.randomUUID().toString();
+    var creationParameters =
+        new GcsBucketCreationParameters()
+            .name(bucketName)
+            .location(BUCKET_LOCATION)
+            .defaultStorageClass(GcsBucketDefaultStorageClass.STANDARD)
+            .lifecycle(new GcsBucketLifecycle().rules(Collections.emptyList()));
+
+    var privateUser = new PrivateResourceIamRoles();
+    privateUser.add(ControlledResourceIamRole.WRITER);
```

Comment:
It seems odd for this to happen here.

Reply:
`PrivateResourceIamRoles` is really just a name for `ArrayList<ControlledResourceIamRole>`, so it doesn't get the fluent builders like other generated objects. This seemed like the best option